scripts/list_boards.py

```diff
@@ -83,15 +160,74 @@ def find_arch2board_set_in(root, arches):
                 file_name = maybe_defconfig.name
                 if file_name.endswith('_defconfig'):
                     board_name = file_name[:-len('_defconfig')]
-                    ret[arch].add(Board(board_name, arch, maybe_board))
+                    ret[arch].add(Board(board_name, maybe_board, 'v1', arch=arch))
 
     return ret
 
+
+def find_v2_boards(args):
+    root_args = argparse.Namespace(**{'soc_roots': args.soc_roots})
+    systems = list_hardware.find_v2_systems(root_args)
+
+    boards = []
+    board_files = []
+    for root in args.board_roots:
+        board_files.extend((root / 'boards').rglob(BOARD_YML))
+
+    for board_yml in board_files:
+        if board_yml.is_file():
+            with board_yml.open('r') as f:
+                b = yaml.safe_load(f.read())
+
+            try:
+                pykwalify.core.Core(source_data=b, schema_data=board_schema).validate()
+            except pykwalify.errors.SchemaError as e:
+                sys.exit('ERROR: Malformed "build" section in file: {}\n{}'
+                         .format(board_yml.as_posix(), e))
+
+            mutual_exclusive = {'board', 'boards'}
+            if len(mutual_exclusive - b.keys()) < 1:
+                sys.exit(f'ERROR: Malformed content in file: {board_yml.as_posix()}\n'
+                         f'{mutual_exclusive} are mutual exclusive at this level.')
+
+            board_array = b.get('boards', [ b.get('board', None) ])
```

Comment:
In this case it would, but initially, to get rid of the board misuse of making all variants an independent board, then the intention was to only have a single board in a folder, and everything else in that folder should be targeting different SoCs / cores or be a variant of the board.

But some boards has been identified as being real different boards, and not just a form of variant.
Seems to be done like that to share common code.

So allowing multiple boards in one folder is a pragmatic solution for now, but I would like to see if we can enhance the model later so re-usability can be achieved without having to different boards in same folder.

Therefore the preference is still to support:
```
board:
  name: plank
    socs:
    - name: foo
    ...
```
without the need to create a yaml sequence of one.